ServerCore/ModelBases/EventSpecificPageModel.cs

```diff
@@ -39,6 +39,17 @@ public override async Task OnPageHandlerExecutionAsync(PageHandlerExecutingConte
                 return;
             }
 
+            // Check permissions now that the role has been binded
+            await IsEventAdmin();
```

Comment:
I don't think we want to do these checks on every page load even if the requested role is "play". Seems like a perf issue.

Reply:
These are cached, so I can make sure we always use the cached version if it exists, but we definitely need to actually check that the role is valid.  We haven't been doing that at all, which is a huge security hole.

Reply:
Oh actually those functions already don't do the database hit if the cached version already exists, so they're good.

Reply:
OK, agree that the role switcher will cache these values, I forgot about that case.